### ISSUE TITLE
Uploading new version of file fails when IngestJob is performed later

### ISSUE THREAD

**Issue by dchandekstark** [samvera/hyrax#4747]
### Descriptive summary

Hyrax 2.9.3

Traceback from failed IngestJob:

```
RuntimeError
path '/tmp/RackMultipart20210218-169-l21jie.txt' was unusable and uploaded_file empty
/usr/local/bundle/gems/hyrax-2.9.3/app/models/job_io_wrapper.rb:92:in `file_from_uploaded_file!'
/usr/local/bundle/gems/hyrax-2.9.3/app/models/job_io_wrapper.rb:86:in `file'
/usr/local/bundle/gems/hyrax-2.9.3/app/models/job_io_wrapper.rb:25:in `size'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/file.rb:181:in `ldp_headers'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/file_persistence.rb:22:in `block in _update_record'
/usr/local/bundle/gems/ldp-1.0.1/lib/ldp/resource.rb:100:in `block in update'
/usr/local/bundle/gems/ldp-1.0.1/lib/ldp/client/methods.rb:93:in `block (2 levels) in put'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/connection.rb:384:in `block in run_request'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/connection.rb:398:in `block in build_request'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/request.rb:26:in `block in create'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/request.rb:25:in `tap'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/request.rb:25:in `create'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/connection.rb:394:in `build_request'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/connection.rb:379:in `run_request'
/usr/local/bundle/gems/faraday-0.17.3/lib/faraday/connection.rb:175:in `put'
/usr/local/bundle/gems/ldp-1.0.1/lib/ldp/client/methods.rb:89:in `block in put'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/notifications.rb:170:in `instrument'
/usr/local/bundle/gems/ldp-1.0.1/lib/ldp/client/methods.rb:87:in `put'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/caching_connection.rb:24:in `put'
/opt/ruby/lib/ruby/2.6.0/delegate.rb:83:in `method_missing'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/initializing_connection.rb:42:in `put'
/usr/local/bundle/gems/ldp-1.0.1/lib/ldp/resource.rb:98:in `update'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/file_persistence.rb:21:in `_update_record'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/callbacks.rb:249:in `block in _update_record'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:98:in `run_callbacks'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:816:in `_run_update_callbacks'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/callbacks.rb:249:in `_update_record'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/persistence.rb:179:in `create_or_update'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/callbacks.rb:241:in `block in create_or_update'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:98:in `run_callbacks'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:816:in `_run_save_callbacks'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/callbacks.rb:241:in `create_or_update'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/file.rb:166:in `create_or_update'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/with_metadata.rb:22:in `create_or_update'
/usr/local/bundle/gems/active-fedora-12.1.1/lib/active_fedora/persistence.rb:40:in `save'
/usr/local/bundle/gems/hydra-works-1.2.0/lib/hydra/works/services/add_file_to_file_set.rb:45:in `persist'
/usr/local/bundle/gems/hydra-works-1.2.0/lib/hydra/works/services/add_file_to_file_set.rb:35:in `update'
/usr/local/bundle/gems/hydra-works-1.2.0/lib/hydra/works/services/add_file_to_file_set.rb:19:in `call'
/usr/local/bundle/gems/hyrax-2.9.3/app/actors/hyrax/actors/file_actor.rb:25:in `ingest_file'
/usr/local/bundle/gems/hyrax-2.9.3/app/models/job_io_wrapper.rb:68:in `ingest_file'
/opt/app-root/app/jobs/ingest_job.rb:17:in `perform'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/execution.rb:39:in `block in perform_now'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:109:in `block in run_callbacks'
/usr/local/bundle/gems/i18n-1.8.9/lib/i18n.rb:314:in `with_locale'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `instance_exec'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/logging.rb:26:in `block (4 levels) in <module:Logging>'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/notifications.rb:168:in `block in instrument'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/notifications.rb:168:in `instrument'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/logging.rb:25:in `block (3 levels) in <module:Logging>'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/logging.rb:46:in `block in tag_logger'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/tagged_logging.rb:71:in `block in tagged'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/tagged_logging.rb:28:in `tagged'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/tagged_logging.rb:71:in `tagged'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/logging.rb:46:in `tag_logger'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/logging.rb:22:in `block (2 levels) in <module:Logging>'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `instance_exec'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:136:in `run_callbacks'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/execution.rb:38:in `perform_now'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/execution.rb:24:in `block in execute'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:109:in `block in run_callbacks'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/railtie.rb:28:in `block (4 levels) in <class:Railtie>'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/execution_wrapper.rb:87:in `wrap'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/reloader.rb:73:in `block in wrap'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/execution_wrapper.rb:87:in `wrap'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/reloader.rb:72:in `wrap'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/railtie.rb:27:in `block (3 levels) in <class:Railtie>'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `instance_exec'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/usr/local/bundle/gems/activesupport-5.2.4.5/lib/active_support/callbacks.rb:136:in `run_callbacks'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/execution.rb:22:in `execute'
/usr/local/bundle/gems/activejob-5.2.4.5/lib/active_job/queue_adapters/resque_adapter.rb:47:in `perform'
```

### Rationale

This appears to be a bug (regression?) in existing functionality.

### Expected behavior

The IngestJob created by the request succeeds, creating the new file version.

### Actual behavior

The `IngestJob` fails and the new file version is not created.  Comparing this process to creating a new file (not a new version of an existing file), it seems significant that here a `Hyrax::UploadedFile` is *not* persisted in the request.  Instead, the uploaded file path (presumably a `ActionDispatch::Http::UploadedFile`) is given to the `JobIoWrapper` instance that is used in the subsequent `IngestJob`.  Since the temporary upload file only persists during the request lifecycle, the process can be made to succeed by altering the `Hyrax::FileSetActor#update_content` method to run the `IngestJob` synchronously (https://github.com/samvera/hyrax/blob/v2.9.3/app/actors/hyrax/actors/file_set_actor.rb#L45) -- or presumably by using a synchronous ActiveJob adapter.

### Steps to reproduce the behavior

1. Run Hyrax either in a) production environment (i.e., RAILS_ENV=production), or b) Background job processing is configured to run asynchronously with a usual adapter (not `:async`).  We use Resque.
2. Attempt to upload a new version of an existing file through the Hyrax UI.

### Related work

#4052 - This issue was closed as non-reproducible, but I wonder whether it was tested with a production/asynchronous job configuration.

**Comment by no-reply** [comment]
Just dropping this here for reference, since I looked it up: 

nurax (tested in the older ticket) is running sidekiq for background jobs. https://github.com/curationexperts/nurax/blob/master/config/environments/production.rb#L21

**Comment by no-reply** [comment]
@dchandekstark my best guess is that this is a configuration issue in your app. you mentioned in slack that you use an environment variable for configuration. can you confirm that it is propogating to the relevant hyrax configs on both the web server and the rescue workers?

**Comment by dchandekstark** [comment]
@no-reply Thanks for your help.  The Hyrax config in app and resque is identical.  I think the question is why the update request does not persist a Hyrax::UploadedFile, since that seems to me the (or a) way that the expected behavior would work.  The job fails because the path recorded in the JobIoWrapper record, while present when created, is not present at job execution time, unless it is performed sychronously (i.e., during the request cycle).  I will look in more detail through our code as I am not intimately familiar with it (or recent Hyrax generally).  Since reports are that this functionality is not in fact broken, there may well be some factor we introduced.  We did override IngestJob, but the change was slight and not recent; the upstream version has not changed recently either.
